cabal.project

```diff
@@ -16,7 +16,9 @@ index-state:
   , cardano-haskell-packages 2023-09-14T16:16:59Z
 
 packages:
-  hydra-auction-plutus
+  hydra-auction-error
+  hydra-auction-offchain
+  hydra-auction-onchain
```

Comment:
Why?

Reply:
I want to impose a fairly strict separation:

- `hydra-auction-onchain` uses only `plutus` and not `cardano-api`, with very few isolated unavoidable exceptions.
- `hydra-auction-offchain` uses only `cardano-api` and not `plutus`, with very few isolated unavoidable exceptions.
- `hydra-auciton-error` uses neither `plutus` nor `cardano-api`. It defines error types that can be re-used by both on-chain and off-chain. Also, error types need to be in separate modules because I splice error codes into onchain code via template haskell (which requires everything to be imported from outside the module).